### ISSUE TITLE
`Reflect` is only for Proxy traps

### ISSUE THREAD

**Issue by ljharb** [ev1stensberg/proposal-reflect-or#5]
The committee always intended `Reflect` to only contain things that were `Proxy` traps. This is why when the `[[Enumerate]]` `Proxy` trap was removed, `Reflect.enumerate` had to be removed with it.

**Comment by ev1stensberg** [comment]
Could we simply just make this `Object.prototype.logicalAND()` and `Object.prototype.logicalOR()` instead then? Yeah, I understand, but Reflect sounded like a decent target instead of `Object.prototype`, mainly cause the name said so much as well as math calculations was a breeze with `Reflect`. Regardless, do you think this could be something?

**Comment by ljharb** [comment]
No, nothing new can or will be added to `Object.prototype` - that's a non-starter.

**Comment by ljharb** [comment]
I'm not sure I understand the point of this proposal. `var foo = foo || bar` is a broken and legacy pattern anyways, due to the var declaration being hoisted - you'd want `var foo = bar || baz`, sure, but then what's the advantage to a function? you can just use an arrow function inline, with `||`, if you need that.

**Comment by ev1stensberg** [comment]
Indeed, I meant `var foo = bar || baz`. The advantage is that we'd get a more coherent way of declaring `var brandon = brandon || hodor `. I think that it shows much more about what the intended use case for `brandon` would be here. Sorry again, I meant having this as a method on the object constructor. I'd accept the argument that  `var foo = bar || baz` is easy to understand as it is, but when it comes to adding, say an object, like this:  `var option = option || { }` , its getting increasingly more unclear what the purpose was for this obj vs using
 ```js 
var option = ReflectOr(this, {}, null)
```

this, would refer to the object itself, so if you are "Reflecting" upon your own object, it will refer to `option` in this example. If you want to do   `var foo = bar || baz` , you do: 

```js
var foo = ReflectOr(bar, baz, undefined)
```

**Comment by ljharb** [comment]
Again, repeating the LHS on the RHS is an antipattern that shouldn't be done, recommended, or used as a justifying use case.

I'm not sure I agree that it's unclear. `a || b || c` is very explicitly and clearly saying "a if it's truthy, else b if it's truthy, else c". What clarity does the function add to that?

**Comment by ev1stensberg** [comment]
Hi! Sorry for late reply. Been moving out of my appartement and got crazy alot to do with exams these days. 

Yeah, that is true, but it is often use to hydrate an object or provide a fallback value before hydration, and this could possibly ( in my eyes ) somewhat clarify the use case. I also think it's much cleaner, as a part of the Meta-programming modern JavaScript favors( Correct me if I'm wrong on this one ). Regardless, its more expressive in terms of readability and option as well! 

I like to compare this to `var foo = Object.keys(bar)` in terms of readability. Here's a discussion I had with the React team a while back about this, hope it gives you somewhat an insight about who and why people use this. [GitHub Link](https://github.com/reactjs/react-router/pull/3330)
